frontend/src/metabase/collections/components/ActionMenu/ActionMenu.tsx

```diff
@@ -50,8 +52,16 @@ function ActionMenu({
   deleteBookmark,
 }: ActionMenuProps) {
   const isBookmarked = bookmarks && getIsBookmarked(item, bookmarks);
-  const isPreviewOptionShown =
+  const canPin = collection.can_write && item.setPinned != null;
+  const canPreview =
     isItemPinned(item) && isItemQuestion(item) && collection.can_write;
+  const canMove =
+    collection.can_write &&
+    item.setCollection != null &&
+    (!isItemCollection(item) || !isPersonalCollection(item));
```

Comment:
I find this line `(!isItemCollection(item) || !isPersonalCollection(item))` a bit hard to understand at first. Then I realized this is essentially
`!(isItemCollection(item) && isPersonalCollection(item))` or `!canNotMove`.

Maybe there's a way to juggle these flags that make them easier to understand. What do you think?

Reply:
Fixed